src/Microsoft.AspNet.Mvc.Core/Filters/AsyncTimeoutAttribute.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Represents an attribute that is used to set the timeout value, in milliseconds, 
+    /// which when elapsed will cause the current request to be aborted.
+    /// </summary>
+    [AttributeUsage(
+        AttributeTargets.Class | AttributeTargets.Method,
+        AllowMultiple = false,
+        Inherited = true)]
+    public class AsyncTimeoutAttribute : Attribute, IAsyncResourceFilter
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="AsyncTimeoutAttribute"/>.
+        /// </summary>
+        /// <param name="duration">The duration in milliseconds.</param>
+        public AsyncTimeoutAttribute(int duration)
+        {
+            if (duration < -1)
+            {
+                throw new ArgumentException(
```

Comment:
This design prevents cancelling built in functions because the feature is not available by default.

I believe we should have an always available cancellation token this triggers

-----Original Message-----
From: "Kiran Challa" <notifications@github.com>
Sent: ‎2/‎25/‎2015 1:17 PM
To: "aspnet/Mvc" <Mvc@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [Mvc] [Fixes #2044] Implement an equivalent toAsyncTimeOutAttribute (#2083)

In src/Microsoft.AspNet.Mvc.Core/Filters/AsyncTimeoutAttribute.cs:
> +    /// </summary>
> +    [AttributeUsage(
> +        AttributeTargets.Class | AttributeTargets.Method,
> +        AllowMultiple = false,
> +        Inherited = true)]
> +    public class AsyncTimeoutAttribute : Attribute, IAsyncResourceFilter
> +    {
> +        /// <summary>
> +        /// Initializes a new instance of <see cref="AsyncTimeoutAttribute"/>.
> +        /// </summary>
> +        /// <param name="duration">The duration in milliseconds.</param>
> +        public AsyncTimeoutAttribute(int duration)
> +        {
> +            if (duration < -1)
> +            {
> +                throw new ArgumentException(
 
—
Reply to this email directly or view it on GitHub.